Highlight/Highlighter.php

```diff
@@ -563,12 +563,40 @@ public function highlightAuto($code, $languageSubset = null)
      * setAutodetectLanguages will turn on autodetection for all supported
      * languages.
      *
+     * @param bool $include_aliase Specify whether language aliases
+     *      should be included as well.
+     *
      * @return array
      *      An array of language names (strings).
      */
-    public function listLanguages()
+    public function listLanguages($include_aliases = false)
     {
+        if ($include_aliases === true ) {
+            return array_merge(self::$languages, array_keys(self::$aliases));
+        }
+
         return self::$languages;
     }
 
+    /**
+     * Returns list of all available aliases for given language name.
+     *
+     * @param string $language Name or alias of language to look-up.
+     *
+     * @return array
+     *      An array of all aliases associated with the requested
+     *      language name language. Passed-in name is included as
+     *      well.
+     */
+    public function getAliasesForLanguage($language)
+    {
+        $language = self::getLanguage($language);
+
+        if ($language->aliases === null) {
+            return [$language->name];
+        }
+        else {
+            return array_merge([$language->name], $language->aliases);
```

Comment:
I'm not 100% sure if Highlight.php works on PHP 5.3 but for consistency with the rest of the file, please use `array()`.

Also, there's no need for the `else` here. Just return the `array_merge(...)`.

Reply:
Hm... When I tried just plain calling the `array_merge`, I got an error because for some languages the `aliases` will be null (in case there are no aliases). I'll double-check it once again (one of the tests added was touching upon this specific case).

Reply:
Oh sorry, I meant this:

```php
if ($boolean) {
    return $something;
}

return $somethingElse;
```

Since there's nothing more in the `else` condition, there's no need for it.

Reply:
Hm... Yeah, I can confirm that the if/else would be needed - unless handling of the `Language::aliases` is changed so it is an empty array in case of no aliases (line number is probably off because this is working copy):

```
1) HighlightTest::testGetAliasesForLanguageWhenLanguageHasNoAliases
Exception: Language error ("array_merge(): Argument #2 is not an array") thrown from /vagrant/Highlight/Highlighter.php:596
```